src/Pulsar.Client/Auth/Oauth2/AuthenticationOauth2.fs

```diff
@@ -81,6 +80,32 @@ type internal AuthenticationOauth2(issuerUrl: Uri, audience: string, privateKey:
             return temp
         }
 
+    let base64DeserializeCreds data =
+        let bytes = Convert.FromBase64String(data)
```

Comment:
this can be written as
```fsharp
data
|> Convert.FromBase64String
|> System.Text.Encoding.UTF8.GetString
|> JsonSerializer.Deserialize<Credentials>
```

Reply:
Done!